demo/server/server_actions/Route.re

```diff
@@ -0,0 +1,81 @@
+// QUESTION: How should we create this manifest automatically?
+let actionsManifest = (id: string) => {
+  switch (id) {
+  | id when Router.demoActionCreateNote == id => Actions.Notes.createRouteHandler
+  | id when Router.demoActionEditNote == id => Actions.Notes.editRouteHandler
+  | id when Router.demoActionDeleteNote == id => Actions.Notes.deleteRouteHandler
+  | id when Router.demoActionSimpleResponse == id => Actions.Samples.simpleResponseRouteHandler
+  | _ => failwith("No action")
+  };
+};
+
+// QUESTION: How should we create this manifest automatically?
+let formDataManifest = (id: string) => {
+  switch (id) {
+  | id when Router.demoActionFormDataSample == id => Actions.Samples.formDataRouteHandler
+  | id when Router.demoActionFormDataServerOnly == id => Actions.Samples.formDataServerOnlyRouteHandler
+  | _ => failwith("No action")
+  };
+};
+
+type actionContent =
+  | FormData(list((string, list((option(string), string)))))
+  | Body(string);
+
+// This handler is agnostic to the server framework
+// The user of the code passes the content defined by the type actionContent and the actionId
+let actionsHandler = (contentType, actionId) => {
+  switch (contentType, actionId) {
+  | (FormData(formData), actionId) =>
+    // Getting the actionId from the formData as next.js does
+    switch (formData, actionId) {
+    // react-server-dom-webpack encode formData and put the first value as model reference E.g.:["$K1"], 1 is the id
+    // QUESTION: Why is it like this? Is there a change to it be other than "$K"? Is there a way to get more ids?
+    | ([(_, [(_, modelId)]), ...formData], Some(actionId)) =>
+      let chunkId = Yojson.Basic.from_string(modelId);
+      let chunkId =
+        chunkId
+        |> Yojson.Basic.Util.to_list
+        |> List.hd
+        |> Yojson.Basic.Util.to_string;
```

Comment:
why is this necessary? modelId should be a string already

Reply:
React passes the model Id as a List: ["$K{ID}"].

To improve the read I'm changing to:
```reason
      let chunkId = Yojson.Basic.from_string(modelId);
      let chunkId = switch (chunkId) {
      | `List([`String(chunkId)]) => chunkId
      | _ => failwith("Invalid chunkId")
      };
```